application/client/src/app/ui/views/toolbar/chart/zoomer/cursor/state.ts

```diff
@@ -0,0 +1,274 @@
+import { InternalAPI } from '@service/ilc';
+import { ViewContainerRef } from '@angular/core';
+import { Chart, IPositionChange } from '@ui/service/toolbar/chart';
+
+enum EChangeKind {
+    move = 'move',
+    left = 'left',
+    right = 'right',
+    undefined = 'undefined',
```

Comment:
you are using the reserved word `undefined`... it could be a source of errors or at least confuse. Please choose another variant-name... like `notInited` or any other